### PR TITLE
Migrate shibbo

### PR DESCRIPTION
**Pull request by mipyykko**
Wonder if I left out anything.

### CODE REVIEW

File: helm/templates/ingress.yaml [hunk 1]
```diff
@@ -53,5 +53,17 @@ spec:
             backend:
               serviceName: {{ $fullName }}-frontend
               servicePort: 80
+          - path: /Shibboleth.sso
+            backend:
+              serviceName: shibbo-test
+              servicePort: 80
+          - path: /Shibboleth.sso
```
Comment (nygrenh) [inline helm/templates/ingress.yaml]: duplicate

File: helm/templates/shibbo/shibbo-service.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  name: shibbo-test
+  labels:
+    app: shibbo-test
+    {{- include "helm.labels" . | nindent 4 }}
+spec:
+  type: NodePort
+  ports:
+    - port: 80
+      targetPort: shibbo-test
+    - port: 443
```
Comment (nygrenh) [inline helm/templates/shibbo/shibbo-service.yml:13]: only one port needed

File: shibbo/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# COPY --chown=node . ./app
+COPY . /app
+
+# COPY shibboleth/shibboleth2.xml /etc/shibboleth/shibboleth2.xml
+# COPY shibboleth/shib.conf /etc/httpd/conf.d/shib.conf
+# COPY shibboleth/certs /etc/pki/tls/certs
+# COPY shibboleth/sign-login.helsinki.fi.crt /etc/shibboleth/sign-login.helsinki.fi.crt
+# COPY shibboleth/httpd.conf /etc/httpd/conf/httpd.conf
+# COPY shibboleth/ssl.conf /etc/httpd/conf.d/ssl.conf
+# COPY shibboleth/attribute-map.xml /etc/shibboleth/attribute-map.xml
```
Comment (nygrenh) [inline shibbo/Dockerfile:30]: commment that mount these if building locally

File: shibbo/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+# USER node
+
+WORKDIR /app
+
+RUN rm -rf shibboleth && rm -rf docs
+RUN npm ci
+
+ENV NODE_ENV=production
+
+EXPOSE 80 443
```
Comment (nygrenh) [inline shibbo/Dockerfile]: expose just one thing, 80

File: shibbo/bin/run.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+CURRENT_DIR="$(dirname "$0")"
+if [ ! -n "$CIRCLE_SHA1" ]; then
+  source "$CURRENT_DIR/copy-config.sh"
+fi
+
+httpd-shibd-foreground &
```
Comment (nygrenh) [inline shibbo/bin/run.sh]: supervisord

File: shibbo/src/server.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  const client = new GraphQLClient(API_URL, { headers: { authorization } })
+
+  try {
+    client.request(VERIFIED_USER_MUTATION, {
+      display_name: displayname,
+      personal_unique_code: schacpersonaluniquecode,
+      organization_id: HY_ORGANIZATION_ID,
+      organization_secret: HY_ORGANIZATION_SECRET,
+    })
+    res.redirect("https://www.mooc.fi/connection_success")
```
Comment (nygrenh) [inline shibbo/src/server.ts]: also support development mode

**Review by nygrenh — CHANGES_REQUESTED** [review]

**Comment by mipyykko** [comment]
`supervisord` now starts both the app and `shibd`. Running processes as different users seems a bit wonky in supervisor, now it's just root for all.

**Review by nygrenh — APPROVED** [review]
Looks fine, but in case something goes wrong, be ready to rollback the deployment (`helm history moocfi`, `helm rollback moocfi id`)

**Pull request merged**